clients/python-wrapper/pylotl/client.py

```diff
@@ -0,0 +1,114 @@
+import base64
+import binascii
+import copy
+from typing import Optional
+import httpx
+
+import lakefs_sdk
+from lakefs_sdk.client import LakeFSClient
+
+from pylotl.config import ClientConfig
+from pylotl.exceptions import NoAuthenticationFound, UnsupportedOperationException
+
+
+class Client:
+    """
+    Wrapper around lakefs_sdk's client object
+    Takes care of instantiating it from the environment
+    """
+
+    _client: LakeFSClient = None
+    _http_client: httpx.Client = None
+    _conf: ClientConfig = None
+    _storage_conf: lakefs_sdk.StorageConfig = None
+
+    def __init__(self, **kwargs):
+        self._conf = ClientConfig(**kwargs)
+        self._client = LakeFSClient(self._conf.get_config())
+
+        # Set up http client
+        config = self._conf.get_config()
+        headers = {}
+        auth = None
+        if config.access_token is not None:
+            headers["Authorization"] = f"Bearer {config.access_token}"
+
+        if config.username is not None and config.password is not None:
+            auth = httpx.BasicAuth(config.username, config.password)
+
+        self._http_client = httpx.Client(headers=headers, auth=auth)
+
+    def __del__(self):
```

Comment:
I don't think this does what you think it does :)

Reply:
Can you explain?